### DISCUSSION TITLE
Using Git Commands Keeps Asking for Password

### DISCUSSION THREAD

**Discussion by drewcodes0001** [community/community#134971] [New to GitHub]
I have generated an SSH key and added the public key to my account. However, when using git commands it keeps asking for the password. I have searched around and the most common mistake was using the HTTPS link. I have verified that in my git config file it is using the SSH link. Any ideas  on what is wrong here?

UPDATE: It started working after I deleted the `.git` folder from the project I was working on and re-clonging the project. I set created the `.bashrc` file from [Working with SSH key passphrases](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/working-with-ssh-key-passphrases) since I didn't have it and copied and pasted the contents from the article. Not sure if that had anything to do with solving the issue. When I ran `ssh -T git@github.com` it asked said something about not having a `bash_login` file and that `.bashrc` would be linked. After that it doesn't ask for the password anymore. Also, to be clear it started working after I deleted the .git file from my project folder. I had already created the .bashrc file before that and it still asked for the password then.

**Comment by ghost** [comment]


**Reply by drewcodes0001** [reply]
When running `ssh -T git@github.com` I get:
```Enter passphrase for key '/c/Users/drewcodes0001/.ssh/id_ed25519':```
Following that I get the success message you showed above. Running `git remote -v` I get: 
``` 
origin  git@github.com:drewcodes0001/azure-resume.git (fetch)
origin  git@github.com:drewcodes0001/azure-resume.git (push)
```

**Reply by drewcodes0001** [reply]
Yes, those are the steps I followed to use a ssh agent to manage my ssh key. That's why I am confused on why it keeps asking.

**Reply by drewcodes0001** [reply]
I got it to work after deleting the `.git` folder from my project and re-cloning the project from github. Just curious does the ssh-agent need to automatically start whenever using github or is the auto-launching ssh-agent optional from the second article you sent?

**Reply by drewcodes0001** [reply]
That is good to know, thanks! I just realized that after restarting my computer and running the first initial git command it asks for the password and then not anymore until the next computer restart. While this is way more convenient than it asking after every git command is there a way for it to not ask at all? I use ssh for a remote machine where I have passphrase protected keys but it never asks for the passphrase. I would like to get that behavior. 

After doing some reading from [How to make git not prompt for passphrase for ssh key?](https://superuser.com/questions/1010542/how-to-make-git-not-prompt-for-passphrase-for-ssh-key) and [Git Bash asks for SSH passphrase on first run](https://stackoverflow.com/questions/53472359/git-bash-asks-for-ssh-passphrase-on-first-run-after-every-pc-reboot) it has to do with the code from [Working with SSH key passphrases](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/working-with-ssh-key-passphrases). If I delete that code snippet git bash will ask me to enter the passphrase after every command. Looking at the comments from the StackOverflow post someone deleted something to get it to stop asking after reset, but I can't figure out what.

**Comment by anakatacv** [comment]
Hi,

You have to set Git config to store your credentials, like this command:

git config --global credential.helper

If you want to do it only for an specific repo, you can delete "--global" argument.

After this Git ask you one more time the credentials, it's just to be stored.

Hope this can help you!
